.travis.yml

```diff
@@ -14,11 +14,11 @@ before_install:
   - 'if [ "x$TRAVIS_NODE_VERSION" = "x0.8" ]; then npm install -g npm@1.4; fi'
 
 before_script:
-  - 'if [ "x$TRAVIS_NODE_VERSION" != "x0.6" ]; then $(npm bin)/jscs .; fi'
+  # we only need to run jscs once per build, so let's conserve a few resources
+  - 'if [ "x$TRAVIS_NODE_VERSION" = "x0.12" ]; then $(npm bin)/jscs .; fi'
```

Comment:
I've seen this somewhere else before. What is the "x" for?

Reply:
This is an ancient shell script coding pattern to defend against the case where the environment variable has an empty string (or doesn't exist).

If executed as shell command, it would crash in that case, because the `[` command (alias for the `test` command) would not be receiving the expected arguments. E.g. it would see `test`, `=`, `"0.12"` and then missing the third parameter (first equal signs and 0.12 as middle, instead of actual value as first, equal sings in the middle, expected as third).

See also:
* <https://stackoverflow.com/q/6852612/319266>
* <https://stackoverflow.com/a/2237103/319266>
* <https://stackoverflow.com/a/2188223/319266>

Long story short, if you know the shell in question is a Bash shell, and are using `[[` instead of `[` then you do not need this hack. There are also a number of other ways to solve it without this hack.